lib/chef/resource/habitat/habitat_sup.rb

```diff
@@ -0,0 +1,311 @@
+
+      property :launcher_version, String,
+      description: "Allows you to choose which version of launcher to install. Defaults to `latest`"
+
+      property :service_version, String, # Windows only
+      description: "Allows you to choose which version of the **_Windows Service_** to install. Defaults to `latest` (Windows Only)"
+
+      property :keep_latest, String,
+      description: "Automatically cleans up old packages. If this flag is enabled, service startup will initiate an uninstall of all previous versions of the associated package. This also applies when a service is restarted due to an update. If a number is passed to this argument, that number of latest versions will be kept. The same logic applies to the Supervisor package `env:HAB_KEEP_LATEST_PACKAGES=1` (This requires Habitat version `1.5.86+`)"
+
+      property :toml_config, [true, false], default: false,
+      description: "Supports using the Supervisor toml configuration instead of passing exec parameters to the service, default is `false`. [reference](https://www.habitat.sh/docs/reference/#supervisor-config)"
+
+      description "Runs a Habitat Supervisor for one or more Habitat Services. It is used in conjunction with `hab_service` which will manage the services loaded and started within the supervisor."
+
+      examples <<~DOC
+      ```ruby
+      # set up with just the defaults
+      hab_sup 'default'
+
+      # Update listen ports and use Supervisor toml config
+      hab_sup 'test-options' do
+        listen_http '0.0.0.0:9999'
+        listen_gossip '0.0.0.0:9998'
+        toml_config true
+      end
+
+      # Use with an on-prem Builder
+      # Access to public builder may not be available
+      hab_sup 'default' do
+        bldr_url 'https://bldr.private.net'
+      end
+
+      # Using update_condition
+      hab_sup 'default' do
+        bldr_url 'https://bldr.private.net'
+        hab_channel 'dev'
+        update_condition 'track-channel'
+      end
+
+      # Provide event_stream_* information
+      hab_sup 'default' do
+        license 'accept'
+        event_stream_application 'myapp'
+        event_stream_environment 'production'
+        event_stream_site 'MySite'
+        event_stream_url 'automate.private.net:4222'
+        event_stream_token 'myawesomea2clitoken='
+        event_stream_cert '/hab/cache/ssl/mycert.crt'
+      end
+
+      # Provide specific versions
+      hab_sup 'default' do
+        bldr_url 'https://bldr.private.net'
+        sup_version '1.5.50'
+        launcher_version '13458'
+        service_version '0.6.0' # WINDOWS ONLY
+      end
+
+      # Set latest version of packages to retain
+      hab_sup 'default' do
+        bldr_url 'https://bldr.private.net'
+        sup_version '1.5.86'
+        launcher_version '13458'
+        service_version '0.6.0' # WINDOWS ONLY
+        keep_latest '2'
+      end
+      ```
+      DOC
+
+      action :run do
```

Comment:
You can document an action like this which feeds the docs site:

```ruby
action :run, description: "This action does this" do
```

Reply:
good to know I was just trying to figure that out. thanks!